### ISSUE TITLE
Dataplane with healthcheck will never go unhealthy if dp goes away while cp is down

### ISSUE THREAD

**Issue by lahabana** [kumahq/kuma#1737]
### Summary

This tested with Universal with postgres.

### Steps To Reproduce

1. Start the stack in tools/e2e/examples/docker-compose with postgres [extra conf for PG](https://gist.github.com/lahabana/d07dd75455fd4ed7441c36bcddc4a633)
2. Start everything: `make deploy/example/docker-compose -C ../../../..`
4. Stop cp: `docker-compose stop kuma-control-plane`
5. Stop backend v2 sidecar: `docker-compose kill kuma-example-backend-v2-sidecar`
6. Restart cp: `docker-compose start kuma-control-plane`
7. See that the backendv2 is still online:
```
kumactl inspect dataplanes --tag version=v2
MESH      NAME                      TAGS                                                       STATUS   LAST CONNECTED AGO   LAST UPDATED AGO   TOTAL UPDATES   TOTAL ERRORS   CERT REGENERATED AGO   CERT EXPIRATION   CERT REGENERATIONS   KUMA-DP VERSION           ENVOY VERSION
default   kuma-example-backend-v2   env=intg kuma.io/service=kuma-example-backend version=v2   Online   6m                   6m                 8               0              never                  -                 0                    1.0.0-rc2-306-g2059c8f2   1.17.1
```

I'd expect the DPs to become unhealthy (quite quickly) when the cp restarts.

**Comment by lahabana** [comment]
I guess we want to avoid doing a DB write on every healthcheck.

Could there maybe a way to add a job on CP startup with a timeout that marks DPs that haven't reconnected after sometime as unhealthy?


Another option that may solve a few lifetime issues is to persist the streamId/InstanceId  in the DB.
This would enable us to actually detect old dataplanes that don't seem to have an active connection to any CP and update their health accordingly.

This would avoid having too many DB writes as it would be:
- a periodic write by each CP instance
- a write on each streamOpen
- a write on each streamClose
- Some reads to update dps

**Comment by lobkovilya** [comment]
Thanks for spotting this, we've noticed the same problem with Zone statutes. The bug happens because of the way we handle `DataplaneInsight.Subscriptions`. Kuma considers the Dataplane is online if it has at least one subscription without `DisconnectTime`.  

It would be great to reduce the number of writes to DB, but I'd start with the simpler fix: every time Kuma creates a new subscription for the Dataplane it must ensure the previous subscriptions have `DisconnectTime`. If there is a subscription without `DisconnectTime`, then Kuma can fill this field.

**Comment by lahabana** [comment]
This was fixed in #2246
